### ISSUE TITLE
Rmagick won't install

### ISSUE THREAD

**Issue by Adam0964** [rmagick-temp/rmagick#80]
I keep getting the following error every time I try to 'bundle install': 

Can't install MagickWand 0.2.0. Can't install MagickWand. Can't find MagickWand.h.

Any clue, anyone?

**Comment by rmagick-temp** [comment]
Can you please provide more info? OS, Ruby version, install using Gem or manual, paste the output.

Moncef

**Comment by Adam0964** [comment]
I' m running Mountain Lion, using Ruby 2.0 and Rails 3.2.13. 
I tried to install rmagick by adding the gem to the Gemfile. This is the output I got after I issued a 'bundle install' command: 

  checking for Magick-config... yes
checking for ImageMagick version >= 6.5.0... yes
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing `MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing`MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing `MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing`MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
checking for wand/MagickWand.h... no

**\* Can't install MagickWand 0.2.0. Can't install MagickWand. Can't find MagickWand.h. ***

**Comment by mmaiza** [comment]
Did you install ImageMagick? You need to have it installed. 

Moncef

**Comment by Adam0964** [comment]
Yes. If you look at the second line of the input, you will see that I did:

```
checking for ImageMagick version >= 6.5.0... yes
```

**Comment by rmagick-temp** [comment]
I do not think that your install of ImageMagick contains all the necessary libraries and headers. Try re-installing like so:

brew remove imagemagick

brew install imagemagick  –build-from-source

Moncef

**Comment by Adam0964** [comment]
I followed you recommendations and here is the outcome of that:

new-host-3:main_app adam0964$ brew remove imagemagick 
Uninstalling /Users/adam0964/.homebrew/Cellar/imagemagick/6.8.0-10...
new-host-3:main_app adam0964$ brew install imagemagick –build-from-source 
Error: No available formula for –build-from-source 
new-host-3:main_app adam0964$ brew install imagemagick 
Warning: It appears you have MacPorts or Fink installed.
Software installed with other package managers causes known problems for
Homebrew. If a formula fails to build, uninstall MacPorts/Fink and try again.
Warning: Building source; cellar of imagemagick's bottle is /usr/local/Cellar
==> Downloading http://downloads.sf.net/project/machomebrew/mirror/ImageMagick-6.8.0-10.tar.gz
Already downloaded: /Library/Caches/Homebrew/imagemagick-6.8.0-10.tar.gz
==> ./configure --disable-osx-universal-binary --without-perl --prefix=/Users/adam0964/.homebrew/Cellar/imag
==> make install
 /Users/adam0964/.homebrew/Cellar/imagemagick/6.8.0-10: 1405 files, 44M, built in 76 seconds

It appears that imagemagick would not install with –build-from-source option but somehow, it managed its way through despite Mackports, which it doesn't seem to like very much.

And when I tried to install RMagick, I got the same error. Here is the output:

 checking for Ruby version >= 1.8.5... yes
checking for gcc-4.2... yes
checking for Magick-config... yes
checking for ImageMagick version >= 6.4.9... yes
checking for HDRI disabled version of ImageMagick... yes
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing `MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing`MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing `MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing`MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found
checking for stdint.h... yes
checking for sys/types.h... yes
checking for wand/MagickWand.h... no

Can't install RMagick 2.13.2. Can't find MagickWand.h.
**\* extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
    --with-opt-dir
    --with-opt-include
    --without-opt-include=${opt-dir}/include
    --with-opt-lib
    --without-opt-lib=${opt-dir}/lib
    --with-make-prog
    --without-make-prog
    --srcdir=.
    --curdir
    --ruby=/Users/adam0964/.rvm/rubies/ruby-2.0.0-p0/bin/ruby
ERROR:  Error installing rmagick:
    ERROR: Failed to build gem native extension.

```
Building has failed. See above output for more information on the failure.
```

**Comment by rmagick-temp** [comment]
Assuming the install of ImageMagick was successful, your environment is not setup correctly. Try the following:

1 – mdfind MagickWand.h: record the directory that contains it, say dir1 (the full path)

2 – mdfind MagickCore.pc: record the directory that contains it, say dir2 (the full path)

3 – If it found both files then do:

C_INCLUDE_PATH=dir1 PKG_CONFIG_PATH=dir2 gem install rmagick
4 – If it did not find the file then ImageMagick did not install properly. At which point we may have to 

Moncef

**Comment by Adam0964** [comment]
Hi Moncef ,

Thanks for the feedback. I got kinda sidetracked for the last few days. I did, however, managed to get to the bottom of the issue and resolve it. Here is how:

Since I’m using MacPorts as my package Manager, I ran the following commands:

1- sudo port uninstall imagemagick
2- sudo port selfupdate
3- sudo port install imagemagick

After that, installing the rmagick gem was a breeze.

For those using homebrew, I hope they’ll find the following resource as invaluable as I have:

Imagemagick installation on a Mac: 

http://sethvargo.com/20110726/install-imagemagick-on-mountain-lion/

Adam.
